packages/@lwc/babel-plugin-component/src/utils.ts

```diff
@@ -112,10 +112,15 @@ function generateError(
     return error;
 }
 
+function incrementMetricCounter(metric: CompilerMetrics, state: LwcBabelPluginPass) {
+    state.opts.instrumentation?.incrementCounter(metric);
```

Comment:
For the babel plugin, `opts` was the only place I could find that would let us have a global state object that could be accessed from any visitor AND we would then have access to once the babel transform finished. Open to alternatives if there is one.

Existing diagnostics don't hit this issue because we just throw them.